specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -928,6 +1007,218 @@
         },
         "upgradeDetails": {
           "$ref": "#/definitions/UpgradeDetails"
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnectionVaultProperties"
+          },
+          "description": "List of private endpoint connection.",
+          "readOnly": true
+        },
+        "privateEndpointStateForBackup": {
+          "description": "Private endpoint state for backup.",
+          "enum": [
+            "None",
+            "Enabled"
```

Comment:
>"Enabled [](start = 12, length = 8)

enabled but no disabled?  Better to be explicit about it I think.  The description should also be very explicit about what this means.  In all other scenarios I've seen enabling private endpoints enables it for the entire resource's operations

Reply:
actually, if this is readonly what is the purpose of these two properties?

Reply:
We have kept ‘None’ instead of disabled as it’s the default state that the customer experiences when PrivateEndpoint scenarios haven’t been explicitly enabled.

The ARM tracked resource in this case – Vaults, is used in two scenarios – backup and siteRecovery. Both scenarios have different Groupids and different implications of PrivateEndpoint being enabled, hence having PrivateEndpointForBackup enabled will not imply PrivateEndpointForSiteRecovery to be enabled as well.

The customer isn’t supposed to set PrivateEndpointStateForBackup or PrivateEndpointStateForSiteRecovery values himself/herself. These get automatically set by the RP when there exists a privateEndpointConnection in the vault for the respective groupId in an ‘Approved’ state